Update change log for v7.6.0-preview.4 (#25243)
PowerShell/PowerShell: CHANGELOG/preview.md

# Preview Changelog

## [7.6.0-preview.4] - 2025-03-27

### Breaking Changes

- Fix `WildcardPattern.Escape` to escape lone backticks correctly (#25211) (Thanks @ArmaanMcleod!)
- Convert `-ChildPath` parameter to `string[]` for `Join-Path` cmdlet (#24677) (Thanks @ArmaanMcleod!)

### Engine Updates and Fixes

- Add `PipelineStopToken` to `Cmdlet` which will be signaled when the pipeline is stopping (#24620) (Thanks @jborean93!)
- Fallback to AppLocker after `WldpCanExecuteFile` (#24912)
- Move .NET method invocation logging to after the needed type conversion is done for method arguments (#25022)
- Fix share completion with provider and spaces (#19440) (Thanks @MartinGC94!)

### General Cmdlet Updates and Fixes

> - Exclude outvariable assignments within the same CommandAst when inferring variables (#25224) (Thanks @MartinGC94!)

Reviewer: ```suggestion
- Exclude `-OutVariable` assignments within the same `CommandAst` when inferring variables (#25224) (Thanks @MartinGC94!)
```

Reviewer 2: Should we use the format rule for PR naming or it is not good for GitHub search?

Reviewer 3: @iSazonov unless something changed with search the inclusion of hyphens/dashes makes it hard for getting good search results for things like cmdlet/function names & parameters like this

- Fix parameter completion when script requirements fail (#17687) (Thanks @MartinGC94!)
- Improve the completion for attribute arguments (#25129) (Thanks @MartinGC94!)
- Fix completion that relies on pseudobinding in script blocks (#25122) (Thanks @MartinGC94!)
- Don't complete duplicate command names (#21113) (Thanks @MartinGC94!)
- Make `SystemPolicy` public APIs visible but non-op on Unix platforms so that they can be included in `PowerShellStandard.Library` (#25051)
- Set standard handles explicitly when starting a process with `-NoNewWindow` (#25061)
- Fix tooltip for variable expansion and include desc (#25112) (Thanks @jborean93!)
- Add type inference for functions without OutputType attribute and anonymous functions (#21127) (Thanks @MartinGC94!)
- Add completion for variables assigned by command redirection (#25104) (Thanks @MartinGC94!)
- Handle type inference for redirected commands (#21131) (Thanks @MartinGC94!)
- Allow empty prefix string in 'Import-Module -Prefix' to override default prefix in manifest (#20409) (Thanks @MartinGC94!)
- Update variable/property assignment completion so it can fallback to type inference (#21134) (Thanks @MartinGC94!)

> - Use Get-Help approach to find 'about_*.help.txt' files with correct locale for completions (#24194) (Thanks @MartinGC94!)

Reviewer: ```suggestion
- Use `Get-Help` approach to find `about_*.help.txt` files with correct locale for completions (#24194) (Thanks @MartinGC94!)
```

<p>We thank the following contributors!</p>
<p>@ArmaanMcleod, @pressRtowin</p>

</summary>

<ul>
<li>Refactor and add comments to <code>CompletionRequiresQuotes</code> to clarify implementation (#25223) (Thanks @ArmaanMcleod!)</li>
<li>Add <code>QuoteCompletionText</code> method to CompletionHelpers class (#25180) (Thanks @ArmaanMcleod!)</li>
<li>Remove CompletionHelpers <code>escape</code> parameter from <code>CompletionRequiresQuotes</code> (#25178) (Thanks @ArmaanMcleod!)</li>
<li>Refactor CompletionHelpers <code>HandleDoubleAndSingleQuote</code> to have less nesting logic (#25179) (Thanks @ArmaanMcleod!)</li>
<li>Make the use of Oxford commas consistent (#25139)(#25140)(Thanks @pressRtowin!)</li>
<li>Move common completion methods to CompletionHelpers class (#25138) (Thanks @ArmaanMcleod!)</li>
<li>Return <code>Array.Empty</code> instead of collection <code>[]</code> (#25137) (Thanks @ArmaanMcleod!)</li>
</ul>

</details>

### Tools

- Check GH token availability for Get-Changelog (#25133)

### Tests

- Add XUnit test for `HandleDoubleAndSingleQuote` in CompletionHelpers class (#25181) (Thanks @ArmaanMcleod!)

### Build and Packaging Improvements

<details>

<summary>

> <p>We thank the following contributors!</p>

Reviewer 4: @adityapatwardhan There are no names listed to thank.

> <p>We thank the following contributors!</p>

Author: updated

# Preview Changelog

## [7.6.0-preview.4] - 2025-03-27

### Breaking Changes

- Fix `WildcardPattern.Escape` to escape lone backticks correctly (#25211) (Thanks @ArmaanMcleod!)

> - Convert `-ChildPath` parameter to `string[]` for `Join-Path` cmdlet (#24677) (Thanks @ArmaanMcleod!)

Reviewer 4: We should add ThreadJob information here. Not truly breaking but best place to mention the change.

Reviewer 4: ```suggestion
- Fix `WildcardPattern.Escape` to escape lone backticks correctly (#25211) (Thanks @ArmaanMcleod!)
- Convert `-ChildPath` parameter to `string[]` for `Join-Path` cmdlet (#24677) (Thanks @ArmaanMcleod!)

PowerShell 7.6-preview.4 includes the following updated modules:

- **Microsoft.PowerShell.ThreadJob** v2.2.0
- **ThreadJob** v2.1.0

The **ThreadJob** was renamed to **Microsoft.PowerShell.ThreadJob** module. There is no difference
in the functionality of the module. To ensure backward compatibility for scripts that use the old
name, the **ThreadJob** v2.1.0 module is a proxy module that points to the
**Microsoft.PowerShell.ThreadJob** v2.2.0.
```

Author: resolved

# Preview Changelog

## [7.6.0-preview.4]

### Breaking Changes

- Fix `WildcardPattern.Escape` to escape lone backticks correctly (#25211) (Thanks @ArmaanMcleod!)
- Convert `-ChildPath` parameter to `string[]` for `Join-Path` cmdlet (#24677) (Thanks @ArmaanMcleod!)

PowerShell 7.6-preview.4 includes the following updated modules:

- **Microsoft.PowerShell.ThreadJob** v2.2.0
- **ThreadJob** v2.1.0

> The **ThreadJob** was renamed to **Microsoft.PowerShell.ThreadJob** module. There is no difference

Reviewer 4: ```suggestion
The **ThreadJob** module was renamed to **Microsoft.PowerShell.ThreadJob**. There is no difference
```
